inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationScopeResolver.java

```diff
@@ -0,0 +1,257 @@
+package rocks.inspectit.oce.core.instrumentation.config;
+
+import lombok.extern.slf4j.Slf4j;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.NameMatcher;
+import net.bytebuddy.matcher.StringMatcher;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import rocks.inspectit.oce.core.config.model.instrumentation.InstrumentationSettings;
+import rocks.inspectit.oce.core.config.model.instrumentation.scope.InstrumentationScopeSettings;
+import rocks.inspectit.oce.core.config.model.instrumentation.scope.MethodMatcherSettings;
+import rocks.inspectit.oce.core.config.model.instrumentation.scope.NameMatcherSettings;
+import rocks.inspectit.oce.core.config.model.instrumentation.scope.TypeScope;
+import rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationScope;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+import static rocks.inspectit.oce.core.config.model.instrumentation.scope.MethodMatcherSettings.AccessModifier;
+
+/**
+ * This class is used to resolve the {@link InstrumentationScope} based on the {@link InstrumentationScopeSettings} contained
+ * in the configuration.
+ */
+@Component
+@Slf4j
+public class InstrumentationScopeResolver {
+
+    /**
+     * Creates the {@link InstrumentationScope} instances based on the {@link InstrumentationScopeSettings} contained in
+     * the given {@link InstrumentationSettings}.
+     *
+     * @param source the configuration used as basis for the {@link InstrumentationScope}s
+     * @return Returns a map containing {@link InstrumentationScope} instances. The keys are representing a unique identifier for each scope.
+     */
+    public Map<String, InstrumentationScope> resolve(InstrumentationSettings source) {
+        Map<String, InstrumentationScope> scopeMap = new HashMap<>();
+
+        if (source != null && source.getScopes() != null) {

```

Comment:
Removed the source null check but I would keep the other one, because I think it is possible that it can be null - when no instrumentation is defined.